src/components/sidebar/splitLinkView/RoutePathSelector.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import Moment from 'moment';
+import { IRoutePath } from '~/models';
+import { Checkbox } from '~/components/controls';
+import Loader, { LoaderSize } from '~/components/shared/loader/Loader';
+import * as s from './routePathSelector.scss';
+
+interface IRoutePathSelectorProps {
+    routePaths: IRoutePath[];
+    selectedIds: string[];
```

Comment:
When this is array, checking checkboxes is slow sometimes when there are a lot of routePaths

Make it as selectedIds: Object and selectedIds[id] = true / undefined/false if it is checked or not?

Reply:
Can test if this solution would be faster

Reply:
Object is also slow, but maybe a bit faster.